temporian/implementation/numpy/data/event.py

```diff
@@ -89,6 +105,165 @@ def schema(self) -> Event:
             sampling=self.sampling.names,
         )
 
+    @staticmethod
+    def from_dataframe(
+        df: pd.DataFrame,
+        index_names: List[str],
+        timestamp_column: str = "timestamp",
+    ) -> "NumpyEvent":
+        """Convert a pandas DataFrame to a NumpyEvent. Supported
+        dtypes are: np.float64, np.float32, np.int64, np.int32, np.datetime64
+        (for timestamp column).
+
+        Args:
+            df: DataFrame to convert to NumpyEvent.
+            index_names: names of the DataFrame columns to be used as index for the event.
+            timestamp_column: Name for timestamp index. Defaults to "timestamp".
+
+        Returns:
+            NumpyEvent: NumpyEvent created from DataFrame.
+
+        Raises:
+            ValueError: If index_names or timestamp_column are not in df columns.
+            ValueError: If a column has an unsupported dtype.
+
+        Example:
+            >>> import pandas as pd
+            >>> from temporian.implementation.numpy.data.event import NumpyEvent
+            >>> df = pd.DataFrame(
+            ...     data=[
+            ...         [666964, 1.0, 740.0],
+            ...         [666964, 2.0, 508.0],
+            ...         [574016, 3.0, 573.0],
+            ...     ],
+            ...     columns=["product_id", "timestamp", "costs"],
+            ... )
+            >>> event = NumpyEvent.from_dataframe(df, index_names=["product_id"])
+        """
+        # check index names and timestamp name are in df columns
+        missing_columns = [
+            column
+            for column in index_names + [timestamp_column]
+            if column not in df.columns
+        ]
+
+        if missing_columns:
+            raise ValueError(
+                f"Missing columns {missing_columns} in DataFrame. "
+                f"Columns: {df.columns}"
+            )
+
+        # check timestamp_column is not on index_names
+        if timestamp_column in index_names:
+            raise ValueError(
+                f"Timestamp column {timestamp_column} cannot be on index_names"
+            )
+
+        # check column dtypes, every dtype should be a key of DTYPE_MAPPING
+        for column in df.columns:
+            # if it's a date, convert it to duration
+            if is_a_date(df[column].dtype.type):
```

Comment:
I would only do an automated conversion for the timestamp column.

Reply:
And what would happen for other columns if the DTYPE is a date? Should we raise an error?